internal/storage/storage.go

```diff
@@ -49,13 +58,136 @@ func (m *MemStorage) GetMetric(typeMetric, nameMetric string) (string, error) {
 	}
 }
 
-func (m *MemStorage) GetTypeMetric(t string) interface{} {
-	switch t {
+func (m *MemStorage) SetJSONGauge(nameMetric string, data *float64) error {
+	m.gauge[nameMetric] = *data
+	if m.UpdateInterval == 0 {
+		m.FileUpdate(models.Metrics{ID: nameMetric, MType: "gauge", Value: data})
+	}
+	return nil
+}
+
+func (m *MemStorage) SetJSONCounter(nameMetric string, data *int64) error {
+	m.counter[nameMetric] += *data
+	t := m.counter[nameMetric]
+	if m.UpdateInterval == 0 {
+		m.FileUpdate(models.Metrics{ID: nameMetric, MType: "counter", Delta: &t})
+	}
+	return nil
+}
+
+func (m *MemStorage) GetJSONMetric(typeMetric models.MetricType, nameMetric string) (models.Metrics, error) {
+	err := errors.New("не найдена метрика")
+	mt := models.Metrics{}
+	switch typeMetric {
 	case "gauge":
-		return reflect.ValueOf(m).Interface().(*MemStorage).gauge
+		v, t := m.gauge[nameMetric]
+		if !t {
+			return mt, err
+		}
+		mt = models.Metrics{MType: typeMetric, ID: nameMetric, Value: &v}
+		return mt, nil
 	case "counter":
+		v, t := m.counter[nameMetric]
+		if !t {
+			return mt, err
+		}
+		mt = models.Metrics{MType: typeMetric, ID: nameMetric, Delta: &v}
+		return mt, nil
+	default:
+		return mt, err
+	}
+}
+
+func (m *MemStorage) GetTypeMetric(t models.MetricType) interface{} {
+	switch t {
+	case models.MetricGauge:
+		return reflect.ValueOf(m).Interface().(*MemStorage).gauge
+	case models.MetricCounter:
 		return reflect.ValueOf(m).Interface().(*MemStorage).counter
 	}
 
 	return nil
 }
+
+func (m *MemStorage) FileUpdate(metric models.Metrics) {
+	sm := m.ReadFile()
+	sm = dataUpdateOrAdd(sm, metric)
+	mos.SaveMetricsToFile(m.fileName, sm)
+}
+func dataUpdateOrAdd(sm []models.Metrics, metric models.Metrics) []models.Metrics {
+	addF := true
+	for _, m := range sm {
+		if m.MType == "gauge" {
+			if m.ID == metric.ID {
+				*m.Value = *metric.Value
```

Comment:
Почему так решил не реализовывать?
```suggestion
				m.Value =metric.Value
```